internal/broker/bind_create.go

```diff
@@ -134,6 +136,20 @@ func (b *BindEndpoint) Bind(ctx context.Context, instanceID, bindingID string, d
 		expirationSeconds = parameters.ExpirationSeconds
 	}
 
+	lastOperation, err := b.operationsStorage.GetLastOperation(instance.InstanceID)
+	if err != nil {
+		return domain.Binding{}, apiresponses.NewFailureResponse(fmt.Errorf("failed to get last operation for instance %s", instanceID), http.StatusInternalServerError, fmt.Sprintf("failed to get last operation %s", instanceID))
+	}
+	if lastOperation.Type == internal.OperationTypeProvision && (lastOperation.State == domain.InProgress || lastOperation.State == domain.Failed) {
+		var message string
+		if lastOperation.State == domain.InProgress {
+			message = fmt.Sprintf("instance %s creation is in progress", instanceID)
+		} else {
+			message = fmt.Sprintf("instance %s creation failed", instanceID)
+		}
+		return domain.Binding{}, apiresponses.NewFailureResponse(fmt.Errorf(message), http.StatusBadRequest, message) // Agreed with Provisioning API team to return 400
+	}
+
```

Comment:
Maybe we should consider adding a test for this new code?

Reply:
In `cmd/broker/binding_test.go` there are two test cases. One for provisioning in progress, the second for failed one.

Reply:
ok, I missed this test file